### PR TITLE
Allow to create docker registry credentials from the form

### PR DESCRIPTION
**Pull request by andresmgot**
### Description of the change

Follow up of #1684

Add the subform to create docker credentials:
Screenshot from 2020-04-21 18-44-05

### Applicable issues

  - ref #1522

### CODE REVIEW

File: dashboard/src/components/Config/AppRepoList/AppRepoAddDockerCreds.test.tsx [hunk 1]
```diff
@@ -46,3 +50,46 @@ it("select secrets", () => {
...
+  const user = "foo";
+  const password = "pass";
+  const email = "foo@bar.com";
+  const server = "docker.io";
+  wrapper.setState({ secretName, user, password, email, server, showSecretSubForm: true });
+
+  const button = wrapper.find("button").filterWhere(a => a.text().includes("Submit"));
+  button.simulate("click");
+
+  await wait(1);
```
Comment (absoludity) [inline dashboard/src/components/Config/AppRepoList/AppRepoAddDockerCreds.test.tsx:85]: As on the previous PR, keen to know why this hasn't been solved elsewhere without a wait.

File: dashboard/src/components/Config/AppRepoList/AppRepoAddDockerCreds.tsx [hunk 2]
```diff
@@ -30,9 +64,148 @@ export class AppRepoAddDockerCreds extends React.Component<IAppRepoFormProps, {}
...
+            <div className="row">
+              <div className="col-1 margin-t-normal">
+                <label htmlFor="kubeapps-docker-cred-server">Server</label>
+              </div>
+              <div className="col-11">
+                <input
+                  id="kubeapps-docker-cred-server"
+                  value={this.state.server}
+                  onChange={this.handleServerChange}
+                  placeholder="docker.io"
```
Comment (absoludity) [inline dashboard/src/components/Config/AppRepoList/AppRepoAddDockerCreds.tsx:92]: Let's use the full correct placeholder for docker.io? https://index.docker.io/v1

File: dashboard/src/shared/Secret.ts [hunk 3]
```diff
@@ -40,6 +40,39 @@ export default class Secret {
...
+          username: user,
+          password,
+          email,
+          auth: btoa(`${user}:${password}`),
+        },
+      },
+    };
+    const { data } = await axiosWithAuth.post<ISecret>(url, {
+      apiVersion: "v1",
+      data: {
```
Comment (absoludity) [inline dashboard/src/shared/Secret.ts:64]: You can also use the write-only `stringData` key and let k8s do the base64 encoding itself (a bit safer, given different encodings etc.)

File: dashboard/src/components/Config/AppRepoList/AppRepoAddDockerCreds.test.tsx (hunk 1 above)
Comment (andresmgot) [reply dashboard/src/components/Config/AppRepoList/AppRepoAddDockerCreds.test.tsx:85]: will reply in the other PR

**Pull request merged**